cpp/include/cudf_test/iterator_utilities.hpp

```diff
@@ -121,15 +121,18 @@ template <typename Iter>
  * The returned iterator yields `false` (to mark `null`) at the indices corresponding to the
  * pointers having `nullptr` values and `true` for the remaining indices.
  *
+ * @note The input vector is referenced by the transform iterator, so the
+ * lifespan must be just as long as the iterator.
+ *
  * @tparam T the data type
  * @param ptrs The data pointers for which the validity iterator is computed
  * @return auto Validity iterator
  */
 template <class T>
 [[maybe_unused]] static auto nulls_from_nullptrs(std::vector<T const*> const& ptrs)
 {
-  // The vector `indices` is copied into the lambda as it can be destroyed at the caller site.
-  return thrust::make_transform_iterator(ptrs.begin(), [ptrs](auto ptr) { return ptr != nullptr; });
+  return thrust::make_transform_iterator(ptrs.begin(),
+                                         [&ptrs](auto ptr) { return ptr != nullptr; });
```

Comment:
Sorry I'm confused: Why do we need to keep `ptrs` in the lambda? It is unused right?

Reply:
Good point, we don't need the capturing of `ptrs`. Removed it.